projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/linked-cases/write-linked-cases-field.component.ts

```diff
@@ -0,0 +1,187 @@
+import { AbstractAppConfig } from '../../../../app.config';
+import { CaseEditDataService } from '../../../commons/case-edit-data';
+import { CaseField, ErrorMessage } from '../../../domain';
+import { CaseView } from '../../../domain/case-view';
+import { CommonDataService } from '../../../services/common-data-service/common-data-service';
+import { CasesService } from '../../case-editor/services/cases.service';
+import { AbstractFieldWriteComponent } from '../base-field';
+import { CaseLink, LinkedCasesState } from './domain';
+import { LinkedCasesErrorMessages, LinkedCasesEventTriggers, LinkedCasesPages } from './enums';
+import { LinkedCasesService } from './services';
+
+@Component({
+  selector: 'ccd-write-linked-cases-field',
+  templateUrl: './write-linked-cases-field.component.html'
+})
+export class WriteLinkedCasesFieldComponent extends AbstractFieldWriteComponent implements OnInit, AfterViewInit {
+
+  public caseEditForm: FormGroup;
+  public caseDetails: CaseView;
+  public linkedCasesPage: number;
+  public linkedCasesPages = LinkedCasesPages;
+  public linkedCasesEventTriggers = LinkedCasesEventTriggers;
+  public linkedCases: CaseLink[] = [];
+  public errorMessages: ErrorMessage[] = [];
+
+  constructor(
+    private readonly appConfig: AbstractAppConfig,
+    private readonly commonDataService: CommonDataService,
+    private readonly casesService: CasesService,
+    private readonly linkedCasesService: LinkedCasesService,
+    private readonly caseEditDataService: CaseEditDataService) {
+    super();
+  }
+
+  public ngOnInit(): void {
+    // Clear validation errors
+    this.caseEditDataService.clearFormValidationErrors();
+    // Get linked case reasons from ref data
+    this.getLinkedCaseReasons();
+    this.linkedCasesService.editMode = false;
+    this.caseEditDataService.caseDetails$.subscribe({
+      next: caseDetails => this.initialiseCaseDetails(caseDetails)
+    });
+    this.caseEditDataService.caseEventTriggerName$.subscribe({
+      next: name => this.linkedCasesService.isLinkedCasesEventTrigger = (name === LinkedCasesEventTriggers.LINK_CASES)
+    });
+    this.caseEditDataService.caseEditForm$.subscribe({
+      next: editForm => this.caseEditForm = editForm
+    });
+  }
+
+  public initialiseCaseDetails(caseDetails: CaseView): void {
+    if (caseDetails) {
+      this.caseDetails = caseDetails;
+      this.linkedCasesService.caseDetails = caseDetails;
+      this.linkedCasesService.caseId = caseDetails.case_id;
+      this.linkedCasesService.caseName = this.linkedCasesService.getCaseName(caseDetails);
+      this.getLinkedCases();
+    }
+  }
+
+  public ngAfterViewInit(): void {
+    let labelField = document.getElementsByClassName('govuk-heading-l');
+    if (labelField && labelField.length) {
+      labelField[0].replaceWith('')
+    }
+    labelField = document.getElementsByClassName('heading-h2');
+    if (labelField && labelField.length) {
+      labelField[0].replaceWith('')
```

Comment:
@johnbenjamin-hmcts - missing semicolon

Reply:
@Daniel-Lam I have updated the code now.